File: iosApp/iosApp/Models/CountdownTimer/CountdownTimer.swift
//
//  CountDownTimer.swift
//  BauchGlück
//
//  Created by Frederik Kohler on 19.07.24.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation

struct CountdownTimer: Identifiable, Codable {
    var id: String
    var userId: String
    var name: String
    var duration: Int
    var startDate: Date?
    var endDate: Date?
    var timerState: String
    var timerType: String
    var remainingDuration: Int
    var notificate: Bool = true
    var showAvtivity: Bool = true
}

File: shared/src/commonMain/kotlin/model/countdownTimer/CountdownTimer.kt
package model.countdownTimer

import DateTime
import dev.gitlive.firebase.firestore.FieldValue
import kotlinx.serialization.Serializable

@Serializable
data class CountdownTimer(
    var id: String = "",
    var userId: String = "",
    var name: String = "",
    var duration: Int = 0,
    var startDate: FieldValue?,
    var endDate: FieldValue?,
    var timerState: String = "",
    var timerType: String = "",
    var remainingDuration: Int = 0,
    var notificate: Boolean = true,
    var showActivity: Boolean = true,
)